### ISSUE TITLE
mstpr-brainbot - If game has only 1 participant then there will be no winner and ETH will be stuck in the game contract

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-10-looksrare-judging#22]
mstpr-brainbot

medium

# If game has only 1 participant then there will be no winner and ETH will be stuck in the game contract
## Summary
The game can commence under two conditions:

1- Once the total supply is reached.
2- When the mintEnd time is reached.
If the game has only one participant, then as soon as the game starts, the first fulfillRandomWords request from Chainlink will eliminate the sole agent, reducing the number of alive agents to zero. With no surviving agents, there will be no prize to claim. Even though the only claimable prize would have been the ETH deposited by that lone agent, the agent's ETH will not be refunded.
## Vulnerability Detail
Assume there are many games played at the same time and one of the games has no participants and only 1 participant which is Alice and she deposited some ETH to bought her only agent. When the game starts because `block.timestamp > mintEnd` the very first chainlink call will kill the only agent which is the Alice's agent. This will make the alive agents as 0 and the ETH Alice deposited is stucked at the game contract. Alice can not call the `claimGrandPrize` to get her ETH back because of this check:
https://github.com/sherlock-audit/2023-10-looksrare/blob/86e8a3a6d7880af0dc2ca03bf3eb31bc0a10a552/contracts-infiltration/contracts/Infiltration.sol#L657

## Impact
Although it is not very likely to begin a game with only 1 agent it is still possible especially if there can be many games played simultaneously. Also, if such case happens owner can use the emergency withdraw to reimburse Alice's ETH. Considering all of that I label this as medium since it requires an additional tx from the owner and also it spends unnecessary LINK tokens when it calls the fulfill request. So there are some funds lost for the owner and the functionality is not fully correct (do not start the game with only 1 agent)
## Code Snippet
https://github.com/sherlock-audit/2023-10-looksrare/blob/86e8a3a6d7880af0dc2ca03bf3eb31bc0a10a552/contracts-infiltration/contracts/Infiltration.sol#L499-L523
https://github.com/sherlock-audit/2023-10-looksrare/blob/86e8a3a6d7880af0dc2ca03bf3eb31bc0a10a552/contracts-infiltration/contracts/Infiltration.sol#L1096-L1249
## Tool used

Manual Review

## Recommendation
Do not let a game start with only 1 agent 

Duplicate of #65

**Comment by mstpr** [comment]
Escalate

If there is only 1 owner emergency withdraw will not work means that the ETH is lost for the agent.

Here the additional code, you can paste this part to the end of the function

```solidity
        uint256 ethBalanceBefore = address(infiltration).balance;
        console.log("ETH balance in infiltration before emergency withdraw", ethBalanceBefore);

        vm.prank(owner);
        infiltration.emergencyWithdraw();

        uint256 ethBalanceAfter = address(infiltration).balance;
        console.log("ETH balance in infiltration after emergency withdraw", ethBalanceAfter);

        assertEq(ethBalanceBefore, ethBalanceAfter);
```

**Comment by sherlock-admin2** [comment]
> Escalate
> 
> If there is only 1 owner emergency withdraw will not work means that the ETH is lost for the agent.
> 
> Here the additional code, you can paste this part to the end of the function
> 
> ```solidity
>         uint256 ethBalanceBefore = address(infiltration).balance;
>         console.log("ETH balance in infiltration before emergency withdraw", ethBalanceBefore);
> 
>         vm.prank(owner);
>         infiltration.emergencyWithdraw();
> 
>         uint256 ethBalanceAfter = address(infiltration).balance;
>         console.log("ETH balance in infiltration after emergency withdraw", ethBalanceAfter);
> 
>         assertEq(ethBalanceBefore, ethBalanceAfter);
> ```

You've created a valid escalation!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by nasri136** [comment]
@mstpr Even though I agree that if the game start the only one alive agent will be killed and ETH will be stuck but as long as `startGame()`  is not called and [36 hours passed](https://github.com/sherlock-audit/2023-10-looksrare/blob/main/contracts-infiltration/contracts/Infiltration.sol#L561) (`conditionThree = currentRoundId == 0 && block.timestamp > uint256(mintEnd).unsafeAdd(36 hours)`) owner can still call `emergencyWithdraw()` and refund the unique minter

**Comment by nevillehuang** [comment]
Agree with @nasri136, see [comment here](https://github.com/sherlock-audit/2023-10-looksrare-judging/issues/65#issuecomment-1802934094).

**Comment by jpopxfile** [comment]
36 hours is not a long time. And if it's only 1 minted, the team can just refund it before 36 hours, and then call emergency withdraw  after 36 hours.

Or the owner could just call premint before starting the game and mint at least 1 more agent, so there is no longer just 1 agent, but more than that 

Overall I don't think this is a medium when there are so many other mitigating factors in place.

**Comment by Evert0x** [comment]
Planning to reject escalation and keep issue state as is

**Comment by Czar102** [comment]
Result:
Low
Duplicate of #65

**Comment by sherlock-admin2** [comment]
Escalations have been resolved successfully!

Escalation status:
- [mstpr](https://github.com/sherlock-audit/2023-10-looksrare-judging/issues/22/#issuecomment-1804473244): rejected
